js_tests/admin/SelectBox.test.js

```diff
@@ -21,3 +21,38 @@ QUnit.test('filter', function(assert) {
     assert.equal($('#id option').length, 1);
     assert.equal($('#id option').text(), "A");
 });
+
+// test for ticket 31522
+// test to check the selectBox move method and also to check
+QUnit.test("move and scrollbar should not reset to the top when moving options from one selectbox to another", function(assert) {
+    const $ = django.jQuery;
+    const fromId = "from_id";
+    const toId = "to_id";
+
+    const optionsCount = 200;
+    $('<select id="' + fromId + '" multiple ></select>').appendTo('#qunit-fixture');
+    $('<select id="' + toId + '" multiple ></select>').appendTo('#qunit-fixture');
+
+    const fromSelectBoxDom = document.getElementById(fromId);
+    const toSelectBoxDom = document.getElementById(toId);
+    const selectedOptionsIndicies = [194, 195, 196, 197, 198, 199];
+    let anOption;
+    for (let i = 0;i < optionsCount;i++) {
+        anOption = new Option((i + 1), (i + 1), false, false);
+        anOption.title = (i + 1);
+        fromSelectBoxDom.appendChild(anOption);
+    }
+    SelectBox.init(fromId);
+    SelectBox.init(toId);
+
+    for(let i = 0;i < selectedOptionsIndicies.length;i++) {
```

Comment:
`for` is a keyword statement, not a function, so space after keyword.

This could iterate using `for ... of`. See #12866.